diff -up st-0.8.2/x.c patch-st-0.8.2/x.c
--- st-0.8.2/x.c	2019-02-09 03:50:41.000000000 -0800
+++ patch-st-0.8.2/x.c	2020-02-04 20:36:57.182068244 -0800
@@ -1446,10 +1446,14 @@ xdrawcursor(int cx, int cy, Glyph g, int
 			g.u = 0x2603;
 		case 0: /* Blinking Block */
 		case 1: /* Blinking Block (Default) */
+			if (IS_SET(MODE_BLINK))
+				break;
 		case 2: /* Steady Block */
 			xdrawglyph(g, cx, cy);
 			break;
 		case 3: /* Blinking Underline */
+			if (IS_SET(MODE_BLINK))
+				break;
 		case 4: /* Steady Underline */
 			XftDrawRect(xw.draw, &drawcol,
 					borderpx + cx * win.cw,
@@ -1458,6 +1462,8 @@ xdrawcursor(int cx, int cy, Glyph g, int
 					win.cw, cursorthickness);
 			break;
 		case 5: /* Blinking bar */
+			if (IS_SET(MODE_BLINK))
+				break;
 		case 6: /* Steady bar */
 			XftDrawRect(xw.draw, &drawcol,
 					borderpx + cx * win.cw,
@@ -1771,6 +1777,8 @@ run(void)
 	int ttyfd;
 	struct timespec drawtimeout, *tv = NULL, now, last, lastblink;
 	long deltatime;
+	int blink_cursor = win.cursor == 0 || win.cursor == 1 ||
+			   win.cursor == 3 || win.cursor == 5;
 
 	/* Waiting for window mapping */
 	do {
@@ -1807,7 +1815,7 @@ run(void)
 		if (FD_ISSET(ttyfd, &rfd)) {
 			ttyread();
 			if (blinktimeout) {
-				blinkset = tattrset(ATTR_BLINK);
+				blinkset = blink_cursor || tattrset(ATTR_BLINK);
 				if (!blinkset)
 					MODBIT(win.mode, 0, MODE_BLINK);
 			}
